contracts/DCNTCrescendo.sol

```diff
@@ -129,8 +133,17 @@ contract DCNTCrescendo is
     return ((bps - takeRateBPS) * _currentPrice[id]) / bps;
   }
 
-  function buy(uint256 id) external payable {
-    require(saleIsActive, "Sale must be active to buy");
+  modifier salesAreActive {
```

Comment:
```suggestion
  modifier onlyActive {
```

I like the naming convention for modifiers as `onlyX` but this is personal preference feel free to ignore.